### PR TITLE
feat: add estimate adjustor to account for gas costs and slippage

### PR DESCRIPTION
**Pull request by ramenforbreakfast**
This adjusts a raw estimate before handing off the to the searching loop to decide if the position size is more profitable than the prior position.

These tests are quite verbose, but are all very similar, mostly just because we need to verify that all the values in the estimate returned are correct. Comments should provide sufficient context.

### CODE REVIEW

File: apps/smart-contracts/core/scripts/ArbitragePools.ts [hunk 1]
```diff
@@ -224,3 +224,62 @@ export const getSlippageFactor = (
   }
   throw new Error('Cannot get estimate for INSUFFICIENT_SPREAD')
 }
+
+/**
+ * `gasCostBuffer` is the Collateral equivalent we are willing to account
+ * for gas fees, e.g. 1 ether would be 1 Collateral in gas cost buffer.
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/scripts/ArbitragePools.ts:230]: nit:
```suggestion
 * `gasCostBuffer` is the Collateral-denominated gas fee to be accounted for
 * e.g. 1 ether would be 1 Collateral in gas cost buffer
```

File: apps/smart-contracts/core/test/integration/ArbitrageTrading.test.ts [hunk 2]
```diff
@@ -740,5 +741,300 @@ describe('=> Arbitrage Trading', () => {
         )
       })
     })
+
+    describe('# getAdjustedEstimate', () => {
+      const TEST_MAX_SLIPPAGE = '0.01' // 1%
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/integration/ArbitrageTrading.test.ts:746]: nit:
```suggestion
      const TEST_MAX_SLIPPAGE = '0.01' // 1%
      
```

**Comment by ramenforbreakfast** [comment]
I have resubmitted simplified tests that focus only on path verification.

**Review by XavierEkkel — APPROVED** [review]
LGTM

**Pull request merged**